File: java/217_Contains_Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        if(nums.length == 0) return false;
        Set<Integer> set = new HashSet<>();
        for(int num: nums){
            if(set.contains(num)){
                return true;
            }
            set.add(num);
        }
        return false;
    }
}

File: scala/217-Contains-Duplicate.scala
import scala.collection.mutable.HashSet

object Solution {
  def containsDuplicate(nums: Array[Int]): Boolean = {
    var seen: HashSet[Int] = HashSet() 
    for (num <- nums) {
      if (seen.contains(num)) return true
      seen.add(num)
    }
    return false
  }
}